stacks_queues/queue.py

```diff
@@ -23,40 +23,65 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            # Queue is full, raise error
+            raise QueueFullException()
+        else:
+            self.rear = (self.rear + 1) % self.buffer_size
+            self.store[self.rear] = element
+            self.size += 1
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException()
+        else:
+            self.front = (self.front + 1) % self.buffer_size
+            front = self.call_front()
+            self.store[self.front] = None
+            self.size -= 1
+            return front
 
-    def front(self):
+    def call_front(self):
         """ Returns an element from the front
             of the Queue and None if the Queue
             is empty.  Does not remove anything.
         """
-        pass
+        return self.store[self.front]
         
 
     def size(self):
         """ Returns the number of elements in
             The Queue
         """
-        pass
+        return self.size
 
     def empty(self):
         """ Returns True if the Queue is empty
             And False otherwise.
         """
-        pass
+        if self.size == 0:
+            return True
+        return False
```

Comment:
Can be a bit more compact
```suggestion
        return self.size == 0
```